src/main/java/io/cdap/plugin/salesforce/plugin/sink/batch/SalesforceSinkConfig.java

```diff
@@ -182,20 +268,28 @@ private void validateSchema(Schema schema) {
     }
   }
 
-  private Set<String> getCreatableSObjectFields() throws ConnectionException {
+  private Set<String> getCreatableSObjectFields(SObjectsDescribeResult describeResult) {
     Set<String> creatableSObjectFields = new TreeSet<>(String.CASE_INSENSITIVE_ORDER);
 
-    AuthenticatorCredentials credentials = this.getAuthenticatorCredentials();
-    PartnerConnection partnerConnection = new PartnerConnection(Authenticator.createConnectorConfig(credentials));
-
-    SObjectDescriptor sObjectDescriptor = SObjectDescriptor.fromName(this.getSObject(), credentials);
-    SObjectsDescribeResult describeResult = new SObjectsDescribeResult(partnerConnection,
-                                                                       sObjectDescriptor.getAllParentObjects());
     for (Field field : describeResult.getFields()) {
       if (field.isCreateable()) {
         creatableSObjectFields.add(field.getName());
       }
     }
     return creatableSObjectFields;
   }
+
+  private SObjectsDescribeResult getSObjectDescribeResult() {
+    AuthenticatorCredentials credentials = this.getAuthenticatorCredentials();
+    PartnerConnection partnerConnection = null;
+    SObjectDescriptor sObjectDescriptor = null;
+    try {
+      partnerConnection = new PartnerConnection(Authenticator.createConnectorConfig(credentials));
```

Comment:
it seems strange to do IO as a side effect in configuration class

Reply:
This is validation logic. It works base on information received from Salesforce server. We validate are the credentials correct, are fields present, are they editable etc. etc.
This is how it works in all other SF plugins as well.